src/components/TextInput/BaseTextInput.js

```diff
@@ -217,6 +219,16 @@ class BaseTextInput extends Component {
                     >
                         <TouchableWithoutFeedback onPress={this.onPress} focusable={false}>
                             <View
+
+                                // When multiline is not supplied, calculating textinput height using onLayout
+                                // eslint-disable-next-line react/jsx-props-no-spreading
+                                {...!this.props.multiline ? {
```

Comment:
Why are we prop spreading here?

Reply:
also instead of a ternary where we return null, we can just do `!this.props.multiline && onLayout` right?

Reply:
@stitesExpensify 
> Why are we prop spreading here?

we are using conditional params that's why, is this format is wrong ?



> also instead of a ternary where we return null, we can just do `!this.props.multiline && onLayout` right?

Yes, we can use like below code

```
{...!this.props.multiline && {
  onLayout: event => (
      this.setState({
          height: event.nativeEvent.layout.height,
      })
  ),
}}
```

Reply:
Does getting rid of the spread not work? So it would just be:

```
{!this.props.multiline && {
  onLayout: event => (
      this.setState({
          height: event.nativeEvent.layout.height,
      })
  ),
}}
```

Reply:
No, this will not work 
![Screenshot 2022-09-02 at 8 52 57 PM](https://user-images.githubusercontent.com/35371050/188182106-ba753f76-c07b-40dc-b20c-9a7fc1bb1fdf.png)

Reply:
@stitesExpensify removed ternary and added && for the condition

Reply:
hm that's weird. Can you explain why that is? It looks like we don't have to spread it here https://github.com/Expensify/App/blob/e254776b8b780e5527c3c31c00a38000f89dc033/src/components/TextInput/BaseTextInput.js#L231

Reply:
> hm that's weird. Can you explain why that is? It looks like we don't have to spread it here
> 
> https://github.com/Expensify/App/blob/e254776b8b780e5527c3c31c00a38000f89dc033/src/components/TextInput/BaseTextInput.js#L231

here we using a condition in rendering that's why we have not used the spread operator, and AFAIK for using conditional props we have to use the spread operator

or we can use this format if we do not want to use the spread operator,

```
onLayout={(event) => this.props.multiline &&
  this.setState({
      height: event.nativeEvent.layout.height,
  })
}
```

Reply:
> > hm that's weird. Can you explain why that is? It looks like we don't have to spread it here
> > https://github.com/Expensify/App/blob/e254776b8b780e5527c3c31c00a38000f89dc033/src/components/TextInput/BaseTextInput.js#L231
> 
> here we using a condition in rendering that's why we have not used the spread operator, and AFAIK for using conditional props we have to use the spread operator
> 
> or we can use this format if we do not want to use the spread operator,
> 
> ```
> onLayout={(event) => this.props.multiline &&
>   this.setState({
>       height: event.nativeEvent.layout.height,
>   })
> }
> ```

@stitesExpensify we can use this if we don't have to use the spread operator

Reply:
@stitesExpensify As @jayeshmangwani mentioned, because props are expected as `object{}`, we need to spread them when we do them conditionally.

@jayeshmangwani I feel what we have is better than what you've recommend in the comment [here](https://github.com/Expensify/App/pull/10455#discussion_r963250248).

Reply:
@mananjadhav Thanks, then we will let it be like that, let's wait for @stitesExpensify  response

Reply:
I agree! Thank you both for explaining for me. Let's use the format without spreading so we don't have to disable a linter line

Reply:
replaced spread operator format

cc: @mananjadhav @stitesExpensify